extensions/box/deployment/src/main/java/org/apache/camel/quarkus/component/box/deployment/BoxProcessor.java

```diff
@@ -0,0 +1,101 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.box.deployment;
+
+import com.box.sdk.BoxAPIRequest;
+import com.box.sdk.BoxDeveloperEditionAPIConnection;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.AdditionalApplicationArchiveMarkerBuildItem;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.SystemPropertyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+import org.apache.camel.component.box.BoxCollaborationsManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxCommentsManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxConfiguration;
+import org.apache.camel.component.box.BoxEventLogsManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxEventsManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxFilesManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxFoldersManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxGroupsManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxSearchManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxTasksManagerEndpointConfiguration;
+import org.apache.camel.component.box.BoxUsersManagerEndpointConfiguration;
+import org.apache.camel.quarkus.core.deployment.UnbannedReflectiveBuildItem;
+
+class BoxProcessor {
+
+    private static final String FEATURE = "camel-box";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    AdditionalApplicationArchiveMarkerBuildItem boxArchiveMarker() {
+        return new AdditionalApplicationArchiveMarkerBuildItem("com/box/sdk");
+    }
+
+    @BuildStep
+    void configure(BuildProducer<SystemPropertyBuildItem> systemProperties,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeClasses) {
+        systemProperties.produce(new SystemPropertyBuildItem("io.netty.noUnsafe", "true"));
```

Comment:
Can 'io.netty.noUnsafe' be a variable?

Reply:
I do not see that box would depend on netty. Is the property really necessary?